storyblok/field-plugin
# feat(cli): deploy options from manifest file

## What?
Deploy options found in the manifest file while creating or updating a plugin through the CLI.

## Why?

JIRA: EXT-1880

This feature will enable the developer to version the plugin's options together with the source code, making it easy to maintain and revert changes.


## How to test? (optional)
1. Navigate to any template (in this example the JS one)

```bash
cd packages/cli/templates/js
```

2. Make adjustments in the `field-plugin.config.json` file (manifest file)

```json
{
    "options": [
        {
            "name": "opt1",
            "value": "val1"
        }
    ]
}
```

3. Create or update a field plugin with the following command

```bash
STORYBLOK_PERSONAL_ACCESS_TOKEN=[your token] npx field-plugin deploy
````

4. Open the generated plugin link and check if the options were sent properly

![Screenshot 2023-10-19 at 19 41 29](https://github.com/storyblok/field-plugin/assets/1240591/e09360db-0687-4933-98d4-7897d53a7877)

![image](https://github.com/storyblok/field-plugin/assets/1240591/1d1be897-dd73-4cf8-9163-3207dc67426c)

I've tested creating and updating plugins for all different kinds of scopes (my-plugins, organization, and partner-portal), I also tested with the skipPrompts parameter. However, any help on testing them would be welcomed 😄
